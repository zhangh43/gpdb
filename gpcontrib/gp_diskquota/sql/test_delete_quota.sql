-- Test delete disk quota
CREATE SCHEMA deleteschema;
SELECT diskquota.set_schema_quota('deleteschema', '1 MB');
SET search_path TO deleteschema;

CREATE TABLE c (i INT);
-- expect failed 
INSERT INTO c SELECT generate_series(1,100000000);
SELECT pg_sleep(5);
-- expect fail
INSERT INTO c SELECT generate_series(1,100);
SELECT diskquota.set_schema_quota('deleteschema', '-1 MB');
SELECT pg_sleep(5);

INSERT INTO c SELECT generate_series(1,100);

DROP TABLE c;
RESET search_path;
DROP SCHEMA deleteschema;
