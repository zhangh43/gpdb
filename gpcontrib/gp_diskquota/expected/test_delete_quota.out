-- Test delete disk quota
CREATE SCHEMA deleteschema;
SELECT diskquota.set_schema_quota('deleteschema', '1 MB');
 set_schema_quota 
------------------
 
(1 row)

SET search_path TO deleteschema;
CREATE TABLE c (i INT);
NOTICE:  Table doesn't have 'DISTRIBUTED BY' clause -- Using column named 'i' as the Greenplum Database data distribution key for this table.
HINT:  The 'DISTRIBUTED BY' clause determines the distribution of data. Make sure column(s) chosen are the optimal data distribution key to minimize skew.
-- expect failed 
INSERT INTO c SELECT generate_series(1,100000000);
ERROR:  schema's disk space quota exceeded with name:deleteschema
SELECT pg_sleep(5);
 pg_sleep 
----------
 
(1 row)

-- expect fail
INSERT INTO c SELECT generate_series(1,100);
ERROR:  schema's disk space quota exceeded with name:deleteschema
SELECT diskquota.set_schema_quota('deleteschema', '-1 MB');
 set_schema_quota 
------------------
 
(1 row)

SELECT pg_sleep(5);
 pg_sleep 
----------
 
(1 row)

INSERT INTO c SELECT generate_series(1,100);
DROP TABLE c;
RESET search_path;
DROP SCHEMA deleteschema;
